drivers/sensor/tmp112/tmp112.c

```diff
@@ -58,20 +46,27 @@ static int tmp112_reg_write(const struct device *dev,
 			       reg, (uint8_t *)&val_be, 2);
 }
 
-static int tmp112_reg_update(const struct device *dev, uint8_t reg,
-			     uint16_t mask, uint16_t val)
+static uint16_t set_config_flags(const struct device *dev, uint16_t mask,
+				 uint16_t value)
 {
-	uint16_t old_val = 0U;
-	uint16_t new_val;
+	struct tmp112_data *data = dev->data;
 
-	if (tmp112_reg_read(dev, reg, &old_val) < 0) {
-		return -EIO;
-	}
+	return (data->config_reg & ~mask) | (value & mask);
+}
+
+static int tmp112_update_config(const struct device *dev, uint16_t mask,
+				uint16_t val)
+{
+	int rc;
+	struct tmp112_data *data = dev->data;
+	const uint16_t new_val = set_config_flags(dev, mask, val);
```

Comment:
`const` redundant here

Reply:
What do you mean ? The `const` is to indicate that `new_val` will not change after being set